applications/asset_tracker_v2/src/events/sensor_module_event.h

```diff
@@ -22,35 +22,65 @@ extern "C" {
 
 #define ACCELEROMETER_AXIS_COUNT 3
 
-/** @brief Sensor event types su bmitted by Sensor module. */
+/** @brief Sensor event types submitted by the Sensor module. */
 enum sensor_module_event_type {
+	/** Motion detected. Acceleration of the device has exceeded the configured threshold.
+	 *  Payload is of type @ref sensor_module_accel_data (accel). The associated acceleration
```

Comment:
Not sure how Doxygen works, but should it be `struct sensor_module_accel_data`? At least that is more self-explaining to me when reading the header file and not the rendered docs. I'd also prefer to mention that `accel` is the struct member that's being referred to, something like `Payload is of type @ref struct sensor_module_accel_data in the 'accel' struct member.`

Reply:
Me neither, I just mimicked how it was done in other libraries. In my understanding, as long as the struct is documented with a brief statement it will be referenced when mentioning it in the API doc. For instance, https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/applications/asset_tracker_v2/doc/cloud_module.html#structcloud__module__data__ack that links to a struct definition in `cloud_module_event.h`.

_I'd also prefer to mention that accel is the struct member that's being referred to, something like Payload is of type @ref struct sensor_module_accel_data in the 'accel' struct member._

That is the point with the (_accel_) addition, but maybe it's not as obvious as I thought.

Reply:
Maybe @peknis has some best practice tips in this case?

Reply:
I'm afraid I don't have any better ideas than what is presented here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/1.8.0/nrf/doc_styleguide.html#references

Reply:
> _I'd also prefer to mention that accel is the struct member that's being referred to, something like Payload is of type @ref struct sensor_module_accel_data in the 'accel' struct member._
> 
> That is the point with the (_accel_) addition, but maybe it's not as obvious as I thought.

Yeah, I know, just didn't find it super clear that what's in the parenthesis refers to the struct member name of the associated payload.
But it's not a big deal, and people will probably understand it when they see the struct definition.